components/server/src/user/user-service.ts

```diff
@@ -248,6 +249,36 @@ export class UserService {
         return AttributionId.render({ kind: "team", teamId: project.teamId });
     }
 
+    async setUsageAttribution(user: User, usageAttributionId: string | undefined): Promise<void> {
```

Comment:
validation in here are the main motivator to externalize `usageAttributionId` from the unchecked `additionalData`.